### PR TITLE
Connection improvements

### PR DESCRIPTION
**Pull request by mrv96**
This PR contains some improvements about instrument connection:

- Added the possibility to open connection implicitly exploiting the with statement.
- Aligned the _set_port_* methods (windows one was ahead the linux one).
- Added the possibility to avoid resource matching. Many times the instrument can be used but for some reason cannot be found by pyvisa resource manager search procedure.
- Added support for TCPIP ports (included SOCKET ports).

### CODE REVIEW

File: easy_scpi/scpi_instrument.py [hunk 1]
```diff
@@ -414,32 +437,29 @@ def _handle_handshake( self ):
...
         port_name = port.upper()
 
-        if (
-            ( not port_name.startswith( 'COM' ) ) and
-            ( not port_name.startswith( 'USB' ) ) and
-            ( not port_name.startswith( 'GPIB' ) )
-        ):
-            raise ValueError( "Port must start with 'COM', 'USB', or 'GPIB'." )
+        if not any(port_name.startswith(p) for p in prefixes):
+            raise ValueError( "Port must start with one of the following: %s." % prefixes )
```
Comment (bicarlsen) [inline easy_scpi/scpi_instrument.py]: Please change to a format string.

File: easy_scpi/scpi_instrument.py [hunk 2]
```diff
@@ -449,14 +469,14 @@ def _set_port_windows( self, port ):
 
         else:
             # redundant error check for future compatibility
-            raise ValueError( "Port must start with 'COM', 'USB', or 'GPIB'." )
+            raise ValueError( "Port must start with one of the following: %s." % prefixes )
```
Comment (bicarlsen) [inline easy_scpi/scpi_instrument.py]: Please use format string.

**Review by bicarlsen — CHANGES_REQUESTED** [review]
Great improvements :) Thank you. I think the only changes are to use format string where indicated. There are also some stylistic changes you could make to adhere to previous styles (spaces between parenthesis and arguments in function calls, etc.) if you're so inclined.

File: easy_scpi/scpi_instrument.py (hunk 1 above)
Comment (mrv96) [reply easy_scpi/scpi_instrument.py]: Solved here: https://github.com/bicarlsen/easy-scpi/pull/8/commits/1ba2d04d9f38a76fb996cb3a9776fd7fa4e763e7

File: easy_scpi/scpi_instrument.py (hunk 2 above)
Comment (mrv96) [reply easy_scpi/scpi_instrument.py]: Resolved here: https://github.com/bicarlsen/easy-scpi/pull/8/commits/1ba2d04d9f38a76fb996cb3a9776fd7fa4e763e7

**Comment by mrv96** [comment]
I've just pushed the changes that you've required. I don't know if i was able to satisfy all coding style rules, by the way, since I made a PR from my main branch to yours, you have the permissions to edit my fork's main branch.

**Review by bicarlsen — APPROVED** [review]

**Pull request merged**